### PR TITLE
Tobin gh690 handle deleted renderpass

### PR DESCRIPTION
**Pull request by tobine**
Fix for #690 
Store creation info for renderpass that Pipeline is created with.

### CODE REVIEW

**Comment by chrisforbes** [comment]
Change looks good; do you have tests?

**Comment by tobine** [comment]
This is fixing a bug with no new checks so I didn't add new tests. RenderPassIncompatible test hits the case that was re-factored. I'm making other changes to add new checks on the deletion of object used by command buffers which will include new tests. I'll make sure to have a deleted renderPass test in that set of changes.

**Comment by chrisforbes** [comment]
I think this may still be broken in the other family of renderpass compatibility functions.

**Comment by tobine** [comment]
Yes, you're correct.  There's a call to validateRenderPassCompatibility() that's still using renderPass from framebuffer createInfo instead of the renderPassCreateInfo that I added to FRAMEBUFFER_NODE. I'll fix that as well.

**Comment by tobine** [comment]
Pushed a follow-on commit. Not pretty, but fixes lifetime issues for now. I'll look at further clean-up in follow-on PRs.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2246,24 +2236,53 @@ static bool verify_renderpass_compatibility(const layer_data *my_data, const VkR
...
+    if (!secondary_render_pass) {
+        stringstream errorStr;
+        errorStr << "invalid VkRenderPass (" << secondaryRP << ")";
+        errorMsg = errorStr.str();
+        return false;
+    }
+    return verify_renderpass_compatibility(dev_data, primaryRPCI, secondary_render_pass->pCreateInfo, errorMsg);
+}
+// For given primary RenderPass object and secondary RPCI, verify that they're compatible
+static bool verify_renderpass_compatibility(const layer_data *dev_data, const VkRenderPass primaryRP,
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Not really happy with growing a pile of overloads of this. Can't we just pull the differences up into the callers? There's ~1 per variant.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -9387,11 +9402,72 @@ static bool validateSubpassCompatibility(layer_data *dev_data, VkCommandBuffer p
...
+        for (uint32_t i = 0; i < primaryPassCI->subpassCount; ++i) {
+            skip_call |= validateSubpassCompatibility(dev_data, primaryBuffer, primaryPassCI, secondaryBuffer, secondaryPassCI, i,
+                                                      primaryPassCI->subpassCount > 1);
+        }
+    }
+    return skip_call;
+}
+
+// Overloaded version that takes primary renderPass CreateInfo and secondary renderPass handle
+static bool validateRenderPassCompatibility(layer_data *dev_data, VkCommandBuffer primaryBuffer,
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Ditto, I think this is crazy.

**Comment by tobine** [comment]
Slightly saner version in #717

**Pull request closed without merging**